hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java

```diff
@@ -308,10 +317,14 @@ private void consolidateInto(
     }
 
     private AccountAmount mergeAdjusts(@NonNull final AccountAmount from, @NonNull final AccountAmount to) {
-        return from.copyBuilder()
-                .amount(from.amount() + to.amount())
-                .isApproval(from.isApproval() || to.isApproval())
-                .build();
+        try {
+            return from.copyBuilder()
+                    .amount(Math.addExact(from.amount(), to.amount()))
+                    .isApproval(from.isApproval() || to.isApproval())
+                    .build();
+        } catch (ArithmeticException e) {
+            throw new IllegalArgumentException("Amount overflow when merging account amounts");
```

Comment:
Since both this and an `ArithmeticException` will propagate to [the same `catch`](https://github.com/hashgraph/hedera-services/blob/main/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/common/AbstractNativeSystemContract.java#L97), probably no value in translating this exception.